lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -1516,4 +1547,50 @@ protected function get_layout_styles( $block_metadata ) {
 		}
 		return $block_rules;
 	}
+
+	/**
+	 * Function that scopes a selector with another one. This works a bit like
+	 * SCSS nesting except the `&` operator isn't supported.
+	 *
+	 * <code>
+	 * $scope = '.a, .b .c';
+	 * $selector = '> .x, .y';
+	 * $merged = scope_selector( $scope, $selector );
+	 * // $merged is '.a > .x, .a .y, .b .c > .x, .b .c .y'
+	 * </code>
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $scope    Selector to scope to.
+	 * @param string $selector Original selector.
+	 * @return string Scoped selector.
+	 */
+	public static function scope_selector( $scope, $selector ) {
+		$scopes    = explode( ',', $scope );
+		$selectors = explode( ',', $selector );
+
+		$selectors_scoped = array();
+		foreach ( $scopes as $outer ) {
+			foreach ( $selectors as $inner ) {
+				$outer = trim( $outer );
+				$inner = trim( $inner );
+				if ( empty( $outer ) ) {
```

Comment:
This is purely a style choice, it's totally up to you. Personally, I find this more readable:

```php
if ( empty( $outer ) && empty( $inner) ) { }
else if ( empty( $outer ) ) {}
else if ( empty( $inner ) ) {}
else {}
```